### ISSUE TITLE
Add `walk` generator to Tree.

### ISSUE THREAD

**Issue by dotsdl** [datreant/datreant.core#52]
It would be a nice convenience to walk a Tree precisely like `scandir.walk` like:

```python
import datreant.core as dtr

for root, dirs, files in dtr.Tree('mytree').walk():
    # do stuff
```

This should be pretty straightforward.

**Comment by jdetle** [comment]
Hi! I am going to get a PR up for this in the upcoming week. (I am posting this to hold myself accountable)

**Comment by dotsdl** [comment]
Hey @jdetle! Welcome to ``datreant``. This is a good issue for getting your feet wet with contribution to the library. Looking forward to your PR!

**Comment by dotsdl** [comment]
Closed by #86.
